### PR TITLE
LaTeX: load amstext in sphinx.sty even if not using amsmath

### PR DESCRIPTION
**Pull request by jfbu**
Needed in particular for:
- use of \iffirstchoice@ at commit 9d82cad (PR #2297)
- use of \text at commit 488ee52 addressing issue #2501

### CODE REVIEW

File: doc/config.rst [hunk 1]
```diff
@@ -1633,6 +1633,17 @@ These options influence LaTeX output.
         .. versionadded:: 1.2
      ``'fontenc'``
         "fontenc" package inclusion, default ``'\\usepackage[T1]{fontenc}'``.
+     ``'amsmath'``
```
Comment (tk0miya) [inline doc/config.rst]: I will remove this in next major release (I haven't done that yet, but I agree what you commented at https://github.com/sphinx-doc/sphinx/commit/d6572259d66cfb9da967f45fb2d671d29c5d5683#commitcomment-17423029)
So, personally, it is better to keep it undocumented.

Comment (jfbu) [inline doc/config.rst]: ok, should I revert this last commit so that this PR does not add documentation for `'amsmath'` key ?

**Comment by tk0miya** [comment]
Yes, please.

**Comment by tk0miya** [comment]
Thanks!

**Comment by jfbu** [comment]
@tk0miya My whole logic looks defective: sphinx/ext/mathbase.py has strong dependency on full amsmath via `equation*`, `split`,  `align`, `aligned`. How should we go about making sure amsmath is loaded if extension sphinx.ext.mathbase is used ? I have no knowledge how extensions are handled. Once this is done the documentation should explain how the user can load other things either before or after. Indeed look at this (head-twisting) excerpt:

> The stix package should be used with at least amsmath v2.14,
> amssymb v3.01 and amsfonts v3.01. With amsmath v2.14 or newer, it
> is recommended to load it (and/or packages that load it) after
> the stix package. Older versions of amsmath must be loaded before
> the stix package, otherwise errors will arise.

from www.ctan.org/pkg/stix.

**Comment by tk0miya** [comment]
At this moment, there is no way to make sure to load amsmath package.
But the `latex_elements['amsmath']` is internal, undocumented key. So we don't worry about that.

**Pull request merged**